src/components/node-list/index.js

```diff
@@ -47,10 +47,11 @@ const NodeListProvider = ({
   onToggleTagFilter,
   onToggleModularPipelineActive,
   onToggleTypeDisabled,
+  onToggleModularPipelineFilter,
   modularPipelines,
 }) => {
   const [searchValue, updateSearchValue] = useState('');
-
+  const [focusMode, setFocusMode] = useState(null);
```

Comment:
In terms of `runFocusMode`, unfortunately I am not sure of the context and have not heard of that feature - is that for experimental tracking? if so I can keep in mind in my next task when I move the focusMode variable to the global state level.

In terms of the difference between the 3, from what I know: 

- `Select` is not a concept widely used within our codebase for variable naming; 

- `Filter` is literally what it is, and is used for most actions / variable naming that relates to the functionality performed under the filter panel. 

- 'Focus' is a relatively new concept which is only introduced with focus mode in this PR - it seems like there is potential additional usage of 'focus' with experimental tracking, which I personally am not aware of as of the details yet when setting this PR up. 

You have brought up a great point in terms of what's pontentially coming up next for focusmode - perhaps it would be great if we can catch up to align on what's in store for focuMode so I can plan my variable naming better for this and my upcoming PR?

Reply:
P.S @limdauto , this local state will actually be removed in my next iteration as I move this state to a global level. I am now aware of the situation with the potential use case of focusmode beyond modular pipelines, which I will name the global state accordingly. In the meantime, to save time in editing back and forth, I will keep the naming as is ( which this variable will not exist in my next iteration.)

Reply:
Thank you for explaining. Yes, happy to keep naming the same in this PR. I will raise this with design as a potential design debt.